datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -53,6 +53,13 @@ abstract class DefaultMatchPattern extends MatchPattern[ModularPlan] {
             (a.child.asInstanceOf[Attribute], a.toAttribute)
           })
 
+    // Create aliasMap with Cast Expression to alias reference attribute
+    val aliasMapCast =
```

Comment:
I think you can merge with above `aliasMap`

Reply:
if aliasMap is changed to Map[Expression, Attribute], then for mv queries having column name and table name as same, fails. hence i think for this case, aliasMap for expression has to be created.